contracts/RestrictedNORI.sol

```diff
@@ -929,7 +926,10 @@ contract RestrictedNORI is
       for (uint256 i = 0; i < ids.length; i++) {
         uint256 id = ids[i];
         Schedule storage schedule = _scheduleIdToScheduleStruct[id];
-        schedule.releasedAmountFloor = _releasedBalanceOfSingleSchedule(id);
+        schedule.releasedAmountFloor = _releasedBalanceOfSingleSchedule(
+          id,
+          _scheduleIdToScheduleStruct[id].releasedAmountFloor
```

Comment:
internally, `_releasedBalanceOfSingleSchedule` calls `_linearReleaseAmountAvailable` which in turn is also loading `Schedule storage schedule = _scheduleIdToScheduleStruct[id]` from storage a second time which seems unnecessary since we can probably just pass the values through from the first load here

Reply:
left a todo for this

Reply:
ah I see... yeah a library would make it so that don't have to pass a schedule struct around as an argument but where we're just calling methods on the schedule struct itself.  going to leave this as a todo for now.  I want these refactors/optimizations to live in their own PR where we can have a specific discussion.